### ISSUE TITLE
`clang-format` and the `--Wno-error=unknown` option

### ISSUE THREAD

**Issue by H-G-Hristov** [microsoft/vscode-cpptools#10072]
### Feature Request

Does the extension support this option:

```
  --Wno-error=<value>            - If set don't error out on the specified warning type.
    =unknown                     -   If set, unknown format options are only warned about.
```

By default `clang-format` will report an error and break any formatting scripts (like the one we use) if it has a unknown option in the config file `.clang-format` like any new options not available in the currently bundled release.

I think this should be an on-off switch which should be enabled by default.

**Comment by H-G-Hristov** [comment]
And example:

```yaml
BasedOnStyle: Google
Standard: c++20
# Available in Clang-Format 15
IndentRequiresClause: true
RequiresClausePosition: OwnLine
# Available in Clang-Format 16
RequiresExpressionIndentation: OuterScope
```

The last line breaks the format request.

**Comment by sean-mcmanus** [comment]
We've added `--Wno-error=unknown` as a default. I've created https://github.com/microsoft/vscode-cpptools/issues/10102 to track adding a setting to make that configurable.

**Comment by sean-mcmanus** [comment]
Fixed with 1.13.4 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.4
